Lab5/README.md

```diff
@@ -2,19 +2,554 @@
+- click on Access policy
+- Select Blob (anonymous read access for blobs only) as the public access level
+
+![public-access-blob-portal][public-access-blob-portal]
+
+# Let's add computer vision to our web application
+
+Azure components are now deployed abd test data is ready. It s now time to add Computer Vision functionality to the web appication.
+
+## Adding the Computer Vision API library to the application
+
+We need to install the Computer Vision client library NuGet package. 
+
+To do so; 
+
+- open the terminal in your Visual Studio Code (```ctrl+` ```) 
+- type ```dotnet add package Microsoft.Azure.CognitiveServices.Vision.ComputerVision```
+
+## Adding the Blob storage library to the application
+
+We need to install the Blob storage library Nuget Package. 
+
+To do so; 
+
+- again in your terminal 
+- type ```dotnet add package Microsoft.Azure.Storage.Blob```
+
+
+
+
+
+## Using the API in code
+
+### Setting up the resources keys
+
+To use the computer vision api, first copy the api key you copied earlier into the file appsettings.json, under the ```Keys:ComputerVision:ApiKey``` section. 
+
+In the ```Keys:ComputerVision:ApiEndPoint``` section, enter the computer vision api endpoint. It should like like this: ```https://eastus.api.cognitive.microsoft.com```, where ```eastus``` is the region into which you created your computer vision api resource into.
+
+Now for the storage connection string, copy the connection string you copied earlier into the section ```Keys:Storage:ConnectionString```.
+
+### Add code to make use of the resource keys
+
+To be able to use these keys into our application, create a class in the application named ```KeysOptions```. We will be using the Options pattern available in ASP.NET Core. In that class, you will need to copy the structure of your appsettings into code. It should look like the following:
+
+```csharp
+public class KeysOptions
+{
+    public ComputerVisionOptions ComputerVision { get; set; }
+    public StorageAccountOptions Storage { get; set; }
+}
+
+public class StorageAccountOptions
+{
+    public string ConnectionString { get; set; }
+}
+
+public class ComputerVisionOptions
+{
+    /// <summary>
+    /// Your subscription key
+    /// </summary>
+    public string ApiKey { get; set; }
+
+    /// <summary>
+    /// The endpoint of the region in which your created your ComputerVision resource. i.e. https://westcentralus.api.cognitive.microsoft.com
+    /// </summary>
+    public string ApiEndPoint { get; set; }
+}
+ ```
+
+ Note that we have also created classes for the sub-sections.
+
+
+
+### Creating classes to consume the storage and computer vision api
+
+Under the root of your application, create a folder called ```Services``` and create 2 classes (2 classes): ```BlobStorageManager``` and ```ImageAnalyzer```
+
+The ```BlobStorageManager``` is used to consume the blob storage and the ```ImageAnalyzer``` is used to consume the computer vision api.
+
```

Comment:
I would add an image (screenshot) of what the folder & files structure should look like once that is done. It will help the developers to validate if he did everything right.

Reply:
done